sklearn/tests/test_base.py

```diff
@@ -992,3 +992,36 @@ def predict(self, X, prop=None):
     with warnings.catch_warnings(record=True) as record:
         CustomOutlierDetector().set_predict_request(prop=True).fit_predict([[1]], [1])
         assert len(record) == 0
+
+
+def test_get_params_html():
+    est = MyEstimator(empty="test")
+
+    assert est._get_params_html() == {"l1": 0, "empty": "test"}
+    assert est._get_params_html().non_default == ["empty"]
+
+
+def test_repr_html_():
+    est = MyEstimator()
+    out = est._get_params_html(deep=False)._repr_html_()
+    assert "<summary>Parameters</summary>" in out
+
+    with config_context(display="text"):
+        msg = "_repr_html_ is only defined when"
+        with pytest.raises(AttributeError, match=msg):
+            est._get_params_html(deep=False)._repr_html_()
+
+
+def test_ReprHTMLMixin_repr_mimebundle():
+    est = MyEstimator()
+    out = est._get_params_html(deep=False)._repr_mimebundle_()
+
+    assert "text/plain" in out
+    assert "text/html" in out
+    assert "<summary>Parameters</summary>" in out["text/html"]
+    assert out["text/plain"] == "{'l1': 0, 'empty': None}"
+
+    with config_context(display="text"):
+        out = est._get_params_html(deep=False)._repr_mimebundle_()
+        assert "text/plain" in out
+        assert "text/html" not in out
```

Comment:
I think that we should not need to test those here. The `test_repr_mimebundle_` specifically for `BaseEstimator` are tested earlier. Here, you are testing the behaviour for `ParamsDict`. So we should instead move those tests in `sklearn/utils/_repr_html/tests/test_params.py` where we will test specifically the behaviour of this class without using the `BaseEstimator._get_params_html` to get the object.